romanisim/regtest/regtestdata.py

```diff
@@ -0,0 +1,490 @@
+    #
+    #     Updates self.input and self.input_remote upon completion
+    #
+    #     Parameters
+    #     ----------
+    #     path: str
+    #         The remote path
+    #
+    #     docopy : bool
+    #         Switch to control whether or not to copy a file
+    #         into the test output directory when running the test.
+    #         If you wish to open the file directly from remote
+    #         location or just to set path to source, set this to `False`.
+    #         Default: `True`
+    #
+    #     get_members: bool
+    #         If an association is the input, retrieve the members.
+    #         Otherwise, do not.
+    #     """
+    #     if path is None:
+    #         path = self.input_remote
+    #     else:
+    #         self.input_remote = path
+    #     if docopy is None:
+    #         docopy = self.docopy
+    #
+    #     # Get the association JSON file
+    #     self.input = get_bigdata(self._inputs_root, self._env, path,
+    #         docopy=docopy)
+    #     with open(self.input) as fp:
+    #         asn = load_asn(fp)
+    #         self.asn = asn
+    #
+    #     # Get each member in the association as well
+    #     if get_members:
+    #         for product in asn['products']:
+    #             for member in product['members']:
+    #                 fullpath = os.path.join(
+    #                     os.path.dirname(self.input_remote),
+    #                     member['expname'])
+    #                 get_bigdata(self._inputs_root, self._env, fullpath,
+    #                             docopy=self.docopy)
+
+    def to_asdf(self, path: str):
+        tree = eval(str(self))
+        af = asdf.AsdfFile(tree=tree)
+        af.write_to(path)
+
+    @classmethod
+    def open(cls, filename: str):
+        with asdf.open(filename) as af:
+            return cls(**af.tree)
+
+
+def run_step_from_dict(rtdata: RegtestData, **step_params) -> RegtestData:
+    """
+    Run Steps with given parameter
+
+    :param rtdata: The artifactory instance
+    :param input_path: The input file path, relative to artifactory
+    :param step: The step to run, either a class or a config file
+    :param args: The arguments passed to `Step.from_cmdline`
+    :returns rtdata: Updated `RegtestData` object with inputs set.
+    """
+
+    # Get the data. If `step_params['input_path]` is not
+    # specified, the presumption is that `rtdata.input` has
+    # already been retrieved.
+    # input_path = step_params.get('input_path', None)
+    # if input_path:
+    #     try:
+    #         rtdata.get_asn(input_path)
+    #     except AssociationNotValidError:
+    #         rtdata.get_data(input_path)
+
+    # Figure out whether we have a config or class
+    step = step_params['step']
+    if step.endswith(('.asdf', '.cfg')):
+        step = os.path.join('config', step)
+
+    # Run the step
+    full_args = [step, rtdata.input]
+    full_args.extend(step_params['args'])
+
+    RomanStep.from_cmdline(full_args)
```

Comment:
I'm not sure how to port over this part of the Roman tests; would the simulator run here?

Reply:
I thought I understood that @nden and you decided that the romancal infrastructure might not be the right template for romanisim in the first place?  She pointed me to 
https://github.com/spacetelescope/simcal/blob/main/simcal/jwst_tests/imaging/test_nircam.py
as an alternative, more stripped-down approach to interacting with stored regression test data.  Has that thinking evolved?

I agree that that code looks pretty attached to the romancal / stpipe 'Step' infrastructure which we don't have in romanisim; I don't immediately know what the equivalent would be.  Maybe we should huddle to touch base about this some time soon?

Reply:
FWIW, just having glanced through that once, I had the impression that things looked lightweight enough that I would be able to put things together---the main new bit for me was going to be to learn how to put files in artifactory.  I was planning to take a swing at this soon and then call for help if I got stuck again.  But you know better than I do how much stuff is going on behind the scenes.